src/angular/components/send/send.component.ts

```diff
@@ -0,0 +1,195 @@
+
+const BroadcasterSubscriptionId = 'SendComponent';
+
+export class SendComponent implements OnInit {
+
+    sendType = SendType;
+    loaded = false;
+    loading = true;
+    refreshing = false;
+    expired: boolean = false;
+    type: SendType = null;
+    sends: SendView[] = [];
+    filteredSends: SendView[] = [];
+    searchText: string;
+    selectedType: SendType;
+    selectedAll: boolean;
+    searchPlaceholder: string;
+    filter: (cipher: SendView) => boolean;
+    searchPending = false;
+
+    actionPromise: any;
+    onSuccessfulRemovePassword: () => Promise<any>;
+    onSuccessfulDelete: () => Promise<any>;
+
+    private searchTimeout: any;
+
+    constructor(protected sendService: SendService, protected i18nService: I18nService,
+        protected platformUtilsService: PlatformUtilsService, protected environmentService: EnvironmentService,
+        protected broadcasterService: BroadcasterService, protected ngZone: NgZone) { }
+
+    async ngOnInit() {
+        this.broadcasterService.subscribe(BroadcasterSubscriptionId, (message: any) => {
+            this.ngZone.run(async () => {
+                switch (message.command) {
+                    case 'syncCompleted':
+                        if (message.successfully) {
+                            await this.load();
+                        }
+                        break;
+                }
+            });
+        });
+
+        await this.load();
+    }
+
+    ngOnDestroy() {
+        this.broadcasterService.unsubscribe(BroadcasterSubscriptionId);
+    }
+
+    async load(filter: (send: SendView) => boolean = null) {
+        this.loading = true;
+        const sends = await this.sendService.getAllDecrypted();
+        this.sends = sends;
+        this.selectAll();
+        this.loading = false;
+        this.loaded = true;
+    }
+
+    async reload(filter: (send: SendView) => boolean = null) {
+        this.loaded = false;
+        this.sends = [];
+        await this.load(filter);
+    }
+
+    async refresh() {
+        try {
+            this.refreshing = true;
+            await this.reload(this.filter);
+        } finally {
+            this.refreshing = false;
+        }
+    }
+
+    async applyFilter(filter: (send: SendView) => boolean = null) {
+        this.filter = filter;
+        await this.search(null);
+    }
+
+    async search(timeout: number = null) {
+        this.searchPending = false;
+        if (this.searchTimeout != null) {
+            clearTimeout(this.searchTimeout);
+        }
+        if (timeout == null) {
+            this.filteredSends = this.sends.filter((s) => this.filter == null || this.filter(s));
+            return;
+        }
+        this.searchPending = true;
+        this.searchTimeout = setTimeout(async () => {
+            this.filteredSends = this.sends.filter((s) => this.filter == null || this.filter(s));
+            this.searchPending = false;
+        }, timeout);
+    }
+
```

Comment:
Missing `addSend()` and `editSend()` methods, or at least interfaces/base impl here? Not sure how modal components or child components would act here, but perhaps those really don't belong in jslib (they may not, just not sure why they're omitted).

Reply:
I originally had them here but empty (with a comment) since it appears almost all actions for both of those will be handled entirely by the platform. What's the better route here?

Reply:
Leaving these methods out for now since the child classes will handle all implementation.